metricbeat/module/system/filesystem/filesystem.go

```diff
@@ -0,0 +1,56 @@
+package filesystem
+
+import (
+	"github.com/elastic/beats/metricbeat/helper"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/topbeat/system"
+)
+
+func init() {
+	helper.Registry.AddMetricSeter("system", "filesystem", New)
+}
+
+// New creates new instance of MetricSeter
+func New() helper.MetricSeter {
+	return &MetricSeter{}
+}
+
+type MetricSeter struct{}
+
+func (m *MetricSeter) Setup(ms *helper.MetricSet) error {
+	return nil
+}
+
+func (m *MetricSeter) Fetch(ms *helper.MetricSet, host string) (event common.MapStr, err error) {
+
+	fss, err := system.GetFileSystemList()
+	if err != nil {
+		logp.Warn("Getting filesystem list: %v", err)
+		return nil, err
+	}
+
+	event = common.MapStr{}
+
+	for _, fs := range fss {
+		fsStat, err := system.GetFileSystemStat(fs)
+		if err != nil {
+			logp.Debug("topbeat", "Skip filesystem %d: %v", fsStat, err)
+			continue
+		}
+		system.AddFileSystemUsedPercentage(fsStat)
+
+		fsEvent := common.MapStr{
+			fsStat.DevName: system.GetFilesystemEvent(fsStat),
```

Comment:
I agree that both implementations should be "almost" identical. I think that topbeat should send the status for processes and file system stats in one event instead of lots of sub events. This still doesn't solve the above problem how to do it in the best way. We could potentially use arrays (https://www.elastic.co/guide/en/elasticsearch/guide/current/complex-core-fields.html#object-arrays) but I have to check how this would work for visualisations.